src/lib/utils/supabase/admin.ts

```diff
@@ -409,6 +409,40 @@ const getUserCredits = async (userId: string) => {
 	return creditData.credits_remaining;
 };
 
+type ProductWithPrices = Product & {
+	prices: Price[];
+	actualPrice?: number;
+};
+
+const getActiveProductsWithPrices = async (limit = 10, offset = 0) => {
+	const {
+		data: products,
+		error,
+		count
+	} = await supabaseAdmin
+		.from('products')
+		.select('*, prices(*)', { count: 'exact' })
+		.eq('active', true)
+		.order('name', { ascending: true })
+		.range(offset, offset + limit - 1);
+
+	if (error) {
+		throw new Error(`Error fetching products: ${error.message}`);
+	}
+
+	const typedProducts: ProductWithPrices[] = products;
+
+	for (const product of typedProducts) {
```

Comment:
Can we cast this instead of declaring a new const
e.g. `products as ProductWithPrices[]`

Reply:
Updated :)